src/Component/Text.jsx

```diff
@@ -0,0 +1,252 @@
+  let messageListItems = messages.map((message) => (
+    <li id={message.id} key={message.id}>
+      <div className="message">
+        <div className="pfp-container">
+          <img src={logo} className="logo" alt="Rocket logo" />
+        </div>
+        <div className="message-content">
+          <img
+            className="message-image"
+            src={message.val.url}
+            alt={message.val.url}
+          />
+          <p>{message.val.textContent}</p>
+        </div>
+        <small className="message-time-stamp">{message.val.time}</small>
+        <Dropdown>
+          <Dropdown.Toggle
+            variant="success"
+            id="dropdownic-basic"
+          ></Dropdown.Toggle>
+          <Dropdown.Menu>
+            <Dropdown.Item
+              onClick={() => {
+                startUpdate(message);
+                document.getElementById("message-input").focus();
+              }}
+            >
+              Edit
+            </Dropdown.Item>
+            <Dropdown.Item onClick={() => deleteMessage(message)}>
+              Delete
+            </Dropdown.Item>
+          </Dropdown.Menu>
+        </Dropdown>
+      </div>
+    </li>
+  ));
+
+  return (
+    <div className="chat" data-bs-theme="dark">
+      <h1 className="chatroom-header">Test Chat</h1>
+      <p className="username"> Hello, {user.email}</p>
+      <div className="chat-container">
+        <div className="messages-container">
+          <ul className="message-list" style={{ listStyle: "none" }}>
+            {messageListItems}
+          </ul>
+          <div className="messages-anchor"></div>
+        </div>
+        <form
+          onSubmit={editing ? editData : writeData}
+          className="text-message-input"
+        >
+          <input
+            autoFocus={true}
+            placeholder="Write message..."
+            type="text"
+            name=""
+            id="message-input"
+            value={textInputValue}
+            onChange={(e) => setTextInputValue(e.target.value)}
+            className="message-field"
+          />
+          <button
+            style={
+              fileInputFile != null
+                ? { border: "1px solid green" }
+                : { border: "none" }
+            }
+            type="button"
+          >
+            <label htmlFor="file-upload">
+              <i className="fa-solid fa-image"></i>
+            </label>{" "}
+            <input
+              type="file"
+              id="file-upload"
+              className="hidden-input"
+              onChange={(e) => {
+                setFileInputFile(e.target.files[0]);
+              }}
+            />
+          </button>
+          <button type="submit">
+            <i className="fa-solid fa-paper-plane"></i>
+          </button>
+        </form>
+        <button onClick={logout}>
+          <i className="fa-solid fa-right-from-bracket"></i>
+        </button>
+      </div>
+    </div>
+  );
+}
```

Comment:
This component could be split further into a form and a display, if you have the time come back and try to achieve this! Otherwise it works well though so good job!